integration_tests/test_mempool.py

```diff
@@ -80,3 +81,53 @@ def test_blocked_address(cronos_mempool):
     rsp = cli.transfer("signer1", cli.address("validator"), "1basecro")
     assert rsp["code"] != 0
     assert "signer is blocked" in rsp["raw_log"]
+
+
+def test_mempool_nonce(cronos_mempool):
+    """
+    test the nonce logic in check-tx after new block is created.
+
+    we'll insert several transactions into mempool with increasing nonces,
+    the tx body is so large that they won't be included in next block at the same time,
+    then we'll try to send a new tx with local nonce to see if it still get accepted even if
+    check-tx state get reset.
```

Comment:
```suggestion
    check-tx state get reset.

    the expected behavior is when mempool.recheck=true, this test will success, because although check-tx state get reset when new blocks generated, but recheck logic will bring it back in sync with pending txs, so the client can keep sending new transactions with local nonce.
```